okurosawa/sels-okurosawa
# Feature/create_dashboard_page

## Description / Purpose (checkpoints)
These change provide feature that User can see all users activities, how many lessons taken and how many words learned. 

These are summary what I did.
- I modified `activity_type` column in `Activities` table because of I want to use `activity_type` as string. 
- I edited `home.blade.php` that shows dashboard. 
- I created components that related to `dashboard page`.
- I added relationship and function to Model.

[*Asana task page*]
https://app.asana.com/0/1173796632051606/1174497130657823

## How To Test This PR
1. Create two accounts.
2. Login to E-Learning system and open dashboard page.
3. You can see all users activities, how many lessons taken and how many words learned.
  But now, there is no function to take lesson or follow other user. So if you want to test about show users activities, you have to insert records to `Lessons` table, `Categories` table, `Words` table, `Answers` table and `Relationships` table using MySQL client software or using SQL directly.

 **Commands**
You have to run this command on your environment because of I added migration file that modify already exists table's column.
```shell
php artisan migrate
```

 **Screenshots**
- Dashboard page if there are some users activities.
![page1](https://user-images.githubusercontent.com/56111888/81539548-5a2fe380-93ab-11ea-8fe0-f4f02d01a6d2.png)

- Dashboard page if there is no users activity.
![page2](https://user-images.githubusercontent.com/56111888/81539712-9c592500-93ab-11ea-8906-f1c3703e782f.png)
